lib/mayaUsd/fileio/writeJobContext.cpp

```diff
@@ -385,6 +385,39 @@ bool UsdMayaWriteJobContext::_NeedToTraverse(const MDagPath& curDag) const
         // If we're not going to create a prim at curDag, then we do not need to
         // traverse.
         return false;
+    } else {
+        // In addition to check for primless, we check for user selection of export types
+        if (mArgs.excludeExportTypes.empty()) {
+            return true;
+        }
+
+        MDagPath shapeDagPath = curDag;
+        if (mArgs.mergeTransformAndShape) {
+            // if we're merging transforms, then we need to look at the shape.
+            shapeDagPath.extendToShape();
+        }
+
+        MStatus                 status;
+        MObject                 obj = shapeDagPath.node();
+        const MFnDependencyNode depFn(obj, &status);
+        if (!status) {
```

Comment:
IDK if failure to cerate a MFnDependencyNode should be treated as failure. A node not being a MFnDependencyNode is not an error, it just that the node does not support the API.

It should just skip the type check.